runtime/doc/lua.txt

```diff
@@ -686,6 +686,19 @@ vim.diff({a}, {b}, {opts})                                        *vim.diff()*
                 Return: ~
                     See {opts.result_type}. nil if {opts.on_hunk} is given.
 
+------------------------------------------------------------------------------
+VIM.MPACK                                                          *lua-mpack*
+
+The *vim.mpack* module provides packing and unpacking of lua objects to
+msgpack encoded strings. |vim.NIL| and |vim.empty_dict()| are supported.
+
+vim.mpack.pack({obj})					*vim.mpack.pack*
```

Comment:
is the "pack"/"unpack" a distinct concept from "encode"/"decode"? We have already `msgpackparse`/`msgpackdump` (Python convention) and `json_ecode`/`json_decode`. 

If the distinction is not important it might make sense to call this `vim.mpack.encode`/decode

Reply:
No it's not, I suspect msgpack libs always use "pack" and "unpack" as it rhymes with "msgpack". I don't think `from_lua` and `to_lua` makes sense as (1) lua is implied by the context of lua code (2) later we need the nouns such as the Packer and the Unpacker, where the Encoder/Decoder works if there's a strong preference to change it to that, though my honest opinion would be to err at the side of lazyness and keep the msgpack conventions.

Reply:
> where the Encoder/Decoder works if there's a strong preference to change it to that, though my honest opinion would be to err at the side of lazyness and keep the msgpack conventions.

if you just want to avoid the chore, I am willing to do it :) Consistency (in the scope of our system) helps users a lot.